hawkular-javaagent/src/main/java/org/hawkular/agent/javaagent/JavaAgentEngine.java

```diff
@@ -235,7 +235,14 @@ public void start() {
 
     @Override
     public void stop() {
-        stopHawkularAgent();
+        // Ensure the JMX stop command returns successfully by doing this in a thread.
+        Runnable r = new Runnable() {
+            @Override
+            public void run() {
+                stopHawkularAgent();
+            }
+        };
+        new Thread(r).start();
```

Comment:
Give a name to this thread, so if it hangs or blows up, the stack trace will give us a clue who called it. i.e:

```
   new Thread(r, "Hawkular Java Agent Stop Thread").start();
```

I also need to think about this some more before merging since we are now making the stop() method asynchronous. We have to check to see what kinds of thread safety/concurrency issues this might cause (especially in cases where the agent is restarted, for example, and the stop is followed quickly with start. We need to make sure nothing in start is initiated before this stop thread is actually kicked off and finishes). In other words, making stop method asynchronous could have some side affects that are "bad" without some synchronization/locking put in place.